editor/limbo_ai_editor_plugin.cpp

```diff
@@ -842,35 +842,80 @@ void LimboAIEditor::_on_history_forward() {
 	EDIT_RESOURCE(history[idx_history]);
 }
 
-void LimboAIEditor::_on_task_dragged(Ref<BTTask> p_task, Ref<BTTask> p_to_task, int p_type) {
-	ERR_FAIL_COND(p_type < -1 || p_type > 1);
-	ERR_FAIL_COND(p_type != 0 && p_to_task->get_parent().is_null());
+void LimboAIEditor::_on_tasks_dragged(const TypedArray<BTTask> &p_tasks, Ref<BTTask> p_to_task, int p_to_pos) {
+	ERR_FAIL_COND(p_to_task.is_null());
+	if (p_tasks.is_empty()) {
+		return;
+	}
 
-	if (p_task == p_to_task) {
+	// Filter tasks
+	Vector<Ref<BTTask>> tasks_list;
+	int no_effect = 0;
+	for (int i = 0; i < p_tasks.size(); i++) {
+		Ref<BTTask> task = p_tasks[i];
+		// Count tasks that don't change position
+		if (task == p_to_task) {
+			if (Math::abs(task->get_index() - p_to_pos) <= 1) {
```

Comment:
Not sure if I follow here. Why is this check necessary? If we're dragging a task onto itself, shouldn't it be skipped without further conditions?

Reply:
Yeah, the signal isn't emitted when this condition is true anyway.